File: web/src/layout/ComponentBleDeviceXml.ts
//
// ComponentBleDeviceXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import htmlForDefault from './component_ble_device.html'
//! Declares com.lightningkite.butterflytemplate.layouts.ComponentBleDeviceXml
export class ComponentBleDeviceXml {
    xmlRoot!: HTMLElement
    deviceName!: HTMLElement
    deviceId!: HTMLElement
    rssi!: HTMLElement
    connect!: HTMLButtonElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.deviceName = getViewById<HTMLElement>(view, "deviceName")
        this.deviceId = getViewById<HTMLElement>(view, "deviceId")
        this.rssi = getViewById<HTMLElement>(view, "rssi")
        this.connect = getViewById<HTMLButtonElement>(view, "connect")
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/ComponentBleDeviceXml.kt
//
// ComponentBleDeviceXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class ComponentBleDeviceXml {

    lateinit var deviceName: TextView
    lateinit var deviceId: TextView
    lateinit var rssi: TextView
    lateinit var connect: Button
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.component_ble_device, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        deviceName = view.findViewById<TextView>(R.id.deviceName)
        deviceId = view.findViewById<TextView>(R.id.deviceId)
        rssi = view.findViewById<TextView>(R.id.rssi)
        connect = view.findViewById<Button>(R.id.connect)
        
        
        
        return view
    }
}